test/trento_web/controllers/v1/profile_controller_test.exs

```diff
@@ -63,4 +63,183 @@ defmodule TrentoWeb.V1.ProfileControllerTest do
 
     assert %{id: ^user_id, fullname: ^fullname, email: ^email} = resp
   end
+
+  test "should get totp enrollment data when totp is not configured for the user", %{
+    conn: conn,
+    api_spec: api_spec
+  } do
+    conn
+    |> put_req_header("content-type", "application/json")
+    |> get("/api/v1/profile/totp_enrollment")
+    |> json_response(:ok)
+    |> assert_schema("UserTOTPEnrollmentPayload", api_spec)
+  end
+
+  test "should get an error when requesting totp enrollment data for an already totp enrolled user",
+       %{
+         conn: conn,
+         api_spec: api_spec
+       } do
+    enrolled_user =
+      insert(:user, %{
+        totp_enabled_at: DateTime.utc_now(),
+        totp_secret: Faker.Internet.domain_name(),
+        totp_last_used_at: DateTime.utc_now()
+      })
+
+    conn =
+      Pow.Plug.assign_current_user(
+        conn,
+        %{"user_id" => enrolled_user.id},
+        Pow.Plug.fetch_config(conn)
+      )
+
+    conn
+    |> put_req_header("content-type", "application/json")
+    |> get("/api/v1/profile/totp_enrollment")
+    |> json_response(:unprocessable_entity)
+    |> assert_schema("UnprocessableEntity", api_spec)
+  end
+
+  test "should return forbidden if the totp enrollment is requested for default admin", %{
+    conn: conn,
+    api_spec: api_spec
+  } do
+    insert(:user, id: 1)
+
+    conn =
+      Pow.Plug.assign_current_user(
+        conn,
+        %{"user_id" => 1},
+        Pow.Plug.fetch_config(conn)
+      )
+
+    conn
+    |> put_req_header("content-type", "application/json")
+    |> get("/api/v1/profile/totp_enrollment")
+    |> json_response(:forbidden)
+    |> assert_schema("Forbidden", api_spec)
+  end
+
+  test "should not reset totp when a reset is requested by the default admin", %{
```

Comment:
This shouldn't be `by the default admin`, it should be `for the default admin`

Reply:
+1